### PR TITLE
Add option to disable job restart

### PR DESCRIPTION
**Pull request by cangencer**
* Change return type of `getFuture()` to `CompletableFuture`
* Various additional Javadoc improvements

### CODE REVIEW

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/JobConfig.java [hunk 1]
```diff
@@ -36,19 +36,59 @@
...
         return splitBrainProtectionEnabled;
     }
 
     /**
-     * Sets split brain protection configuration
+     * Configures the split brain protection feature. When enabled, the job will only
+     * be restarted after a topology change if the quorum size can be met.
+     * The quorum size is calculated as {@code cluster size at job submission time / 2 + 1}.
+     * As a result, the job can only be restarted if the actual cluster size is
+     * bigger than the quorum value.
```
Comment (viliam-durina) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/JobConfig.java]: bigger or equal

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/JobConfig.java [hunk 2]
```diff
@@ -36,19 +36,59 @@
 
     private boolean splitBrainProtectionEnabled;
     private final List<ResourceConfig> resourceConfigs = new ArrayList<>();
+    private boolean autoRestartEnabled = true;
 
     /**
-     * Returns true if split brain protection is enabled
+     * Returns true if {@link #setSplitBrainProtection(boolean)} is enabled.
```
Comment (mtopolnik) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/JobConfig.java]: should be `{@link #setSplitBrainProtection(boolean) split brain protection}`

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/JobConfig.java [hunk 3]
```diff
@@ -36,19 +36,59 @@
...
      */
     public boolean isSplitBrainProtectionEnabled() {
         return splitBrainProtectionEnabled;
     }
 
     /**
-     * Sets split brain protection configuration
+     * Configures the split brain protection feature. When enabled, the job will only
+     * be restarted after a topology change if the quorum size can be met.
+     * The quorum size is calculated as {@code cluster size at job submission time / 2 + 1}.
```
Comment (mtopolnik) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/JobConfig.java]: The net length of a text line should not exceed 72 chars (the initial indentation and `*` are not counted).

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/JobConfig.java (hunk 1 above)
Comment (mtopolnik) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/JobConfig.java]: should be "at least the quorum value"

**Comment by cangencer** [comment]
verify

**Pull request merged**